src/include/sof/common.h

```diff
@@ -206,15 +206,30 @@
 // IS_ENABLED() above.
 #  if XCHAL_HAVE_HIFI5
 #    define SOF_MAX_XCHAL_HIFI 5
+#    define SOF_FRAME_BYTE_ALIGN	16
+#    define SOF_FRAME_BYTE_ALIGN_6CH	16
 #  elif XCHAL_HAVE_HIFI4
 #    define SOF_MAX_XCHAL_HIFI 4
-#  elif XCHAL_HAVE_HIFI3
+#    define SOF_FRAME_BYTE_ALIGN	8
+#    define SOF_FRAME_BYTE_ALIGN_6CH	16
+##  elif XCHAL_HAVE_HIFI3
 #    define SOF_MAX_XCHAL_HIFI 3
+#    define SOF_FRAME_BYTE_ALIGN	8
+#    define SOF_FRAME_BYTE_ALIGN_6CH	16
 #  else
 #    define SOF_MAX_XCHAL_HIFI NONE
 #  endif
 #endif
 
+#if SOF_MAX_XCHAL_HIFI == NONE
+#  ifndef SOF_FRAME_BYTE_ALIGN
+#    define SOF_FRAME_BYTE_ALIGN	4
```

Comment:
or 1?

Reply:
I think one is the default if it's not set, but it makes no sense as align constraint. Also Above I think the #if SOF_MAX_XCHAL_HIFI == NONE could be left out as redundant. These could be before this a macros section for SSE or AVX specific definitions.